bentoml/_internal/configuration/__init__.py

```diff
@@ -28,14 +26,29 @@ class version_mod:
 CONFIG_ENV_VAR = "BENTOML_CONFIG"
 
 
-def expand_env_var(env_var: t.Union[str, bytes]) -> str:
+@t.overload
+def expand_env_var(env_var: None) -> None:
+    ...
+
+
+@t.overload
+def expand_env_var(env_var: str) -> str:
+    ...
+
+
+@t.overload
+def expand_env_var(env_var: t.Union[str, bytes]) -> t.Union[str, bytes]:
+    ...
+
+
+def expand_env_var(
+    env_var: t.Optional[t.Union[str, bytes]]
+) -> t.Optional[t.Union[str, bytes]]:
     """Expands potentially nested env var by repeatedly applying `expandvars` and
     `expanduser` until interpolation stops having any effect.
     """
     if not env_var:
-        raise BentoMLException(
-            f"env_var is expected to be either `str` or `bytes`, not {type(env_var)}"
-        )
+        return env_var
```

Comment:
if it just `str -> str` then this would be unnecessary

Reply:
yup, we can remove that